WooCommerce/Classes/Authentication/Store Creation/StoreCreationCoordinator.swift

```diff
@@ -77,20 +77,22 @@ private extension StoreCreationCoordinator {
         // Disables interactive dismissal of the store creation modal.
         navigationController.isModalInPresentation = true
 
-        if featureFlagService.isFeatureFlagEnabled(.optimizeProfilerQuestions) {
-            navigationController.isNavigationBarHidden = true
+        navigationController.isNavigationBarHidden = true
 
-            showFreeTrialSummaryView(from: navigationController, storeName: WooConstants.defaultStoreName, profilerData: nil)
-        } else {
-            showStoreNameInput(from: navigationController)
-        }
+        showFreeTrialSummaryView(from: navigationController, storeName: WooConstants.defaultStoreName)
     }
 
-    func showProfilerFlow(storeName: String, from navigationController: UINavigationController) {
+    func showProfilerFlow(storeName: String, siteID: Int64, from navigationController: UINavigationController) {
         navigationController.isNavigationBarHidden = false
-        let controller = StoreCreationProfilerQuestionContainerHostingController(viewModel: .init(storeName: storeName, onCompletion: { [weak self] _ in
+        let controller = StoreCreationProfilerQuestionContainerHostingController(viewModel: .init(storeName: storeName, onCompletion: { [weak self] answers in
             guard let self else { return }
-            // TODO: update profiler data with remote
+            // TODO: 10385 Upload profiler answers from `DashboardViewModel`
+            if let answers {
+                self.analytics.track(event: .StoreCreation.siteCreationProfilerData(answers))
+
+                let usecase = StoreCreationProfilerUploadAnswersUseCase(siteID: siteID)
+                usecase.storeAnswers(answers)
```

Comment:
We may need to store the use case strongly in the class as it may be deallocated when this method's done execution, causing the API request to be canceled.

Reply:
The `storeAnswers` is a synchronous method which isn't making any network calls. Do you think we still have to store this `usecase` strongly in a property?

Reply:
If it's not necessary then please feel free to ignore my comment.